libbeat/outputs/fileout/file.go

```diff
@@ -39,7 +40,7 @@ func (out *fileOutput) init(config config) error {
 	out.rotator.Path = config.Path
 	out.rotator.Name = config.Filename
 	if out.rotator.Name == "" {
-		out.rotator.Name = config.Index
+		out.rotator.Name = out.beatName
```

Comment:
Probably best to do this refactoring in a another PR, not to lose the focus on this one.